### ISSUE TITLE
DB lock issue durring unit tests

### ISSUE THREAD

**Issue by smuzaffar** [cms-sw/cmssw#26741]
@ggovi , during PR tests of https://github.com/cms-data/RecoBTag-Combined/pull/23 unit test `testConditionDatabase_2` failed due to locking issue [a]. I think issue is that there are two tests ( https://github.com/cms-sw/cmssw/search?q=cms_conditions_3.db&unscoped_q=cms_conditions_3.db ) which are using `cms_conditions_3.db` file. Can you please provide a fix for this?

[a]
```
===== Test "testConditionDatabase_2" ====
> Connecting with db in sqlite_file:cms_conditions_3.db
CORAL/RelationalPlugins/sqlite Error SQLiteStatement::finalize 5 database is locked
CORAL/RelationalPlugins/sqlite Error SQLiteStatement::execute 5 database is locked
ERROR: database is locked ( CORAL : "SQLiteStatement::execute" from "CORAL/RelationalPlugins/sqlite" )

---> test testConditionDatabase_2 had ERRORS
 
^^^^ End Test testConditionDatabase_2 ^^^^
```

**Comment by smuzaffar** [comment]
assign db

**Comment by fabiocos** [comment]
Any idea why is this appearing just now? This code seems untouched since a while...

**Comment by smuzaffar** [comment]
This is just race condition. If both tests run on same time then there is a chance that there will be lock on the local db file.

**Comment by fabiocos** [comment]
@smuzaffar yes, I understand that, the point is why it starts to systematically hit us now...

**Comment by smuzaffar** [comment]
is it very frequent now?

**Comment by smuzaffar** [comment]
the second test was added on Jan 23 this year ( https://github.com/cms-sw/cmssw/pull/25722 )

**Comment by fabiocos** [comment]
well I would say we are seeing it in several PR tests, while this was not the case until recently. But the IB looks clean from this point of view, are unit tests run there differently?

**Comment by smuzaffar** [comment]
No, unit tests in IBs are run the same way. Only thing is that in IBs we run all  tests that is why there is less chance of collision.

**Comment by smuzaffar** [comment]
https://github.com/cms-sw/cmssw/pull/26762 should fix this issue

**Comment by smuzaffar** [comment]
this has been fixes via #26762
